examples/mnist/pipeline/mnist_dist.py

```diff
@@ -0,0 +1,173 @@
+          loss, global_step=global_step)
+
+      # Test trained model
+      label = tf.argmax(y_, 1, name="label")
+      prediction = tf.argmax(y, 1,name="prediction")
+      correct_prediction = tf.equal(prediction, label)
+
+      accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32), name="accuracy")
+      tf.summary.scalar("acc", accuracy)
+
+      saver = tf.train.Saver()
+      summary_op = tf.summary.merge_all()
+      init_op = tf.global_variables_initializer()
+
+    # Create a "supervisor", which oversees the training process and stores model state into HDFS
+    logdir = TFNode.hdfs_path(ctx, args.model_dir)
+    print("tensorflow model path: {0}".format(logdir))
+    summary_writer = tf.summary.FileWriter("tensorboard_%d" %(worker_num), graph=tf.get_default_graph())
+
+    sv = tf.train.Supervisor(is_chief=(task_index == 0),
+                             logdir=logdir,
+                             init_op=init_op,
+                             summary_op=None,
+                             saver=saver,
+                             global_step=global_step,
+                             stop_grace_secs=300,
+                             save_model_secs=10)
+
+    # The supervisor takes care of session initialization, restoring from
+    # a checkpoint, and closing when done or an error occurs.
+    with sv.managed_session(server.target) as sess:
+      print("{0} session ready".format(datetime.now().isoformat()))
+
+      # Loop until the supervisor shuts down or 1000000 steps have completed.
+      step = 0
+      tf_feed = TFNode.DataFeed(ctx.mgr)
+      while not sv.should_stop() and not tf_feed.should_stop() and step < args.steps:
+        # Run a training step asynchronously.
+        # See `tf.train.SyncReplicasOptimizer` for additional details on how to
+        # perform *synchronous* training.
+
+        # using feed_dict
+        batch_xs, batch_ys = feed_dict(tf_feed.next_batch(batch_size))
+        feed = {x: batch_xs, y_: batch_ys}
+
+        if len(batch_xs) > 0:
+          _, summary, step = sess.run([train_op, summary_op, global_step], feed_dict=feed)
+          # print accuracy and save model checkpoint to HDFS every 100 steps
+          if (step % 100 == 0):
+            print("{0} step: {1} accuracy: {2}".format(datetime.now().isoformat(), step, sess.run(accuracy,{x: batch_xs, y_: batch_ys})))
+
+          if sv.is_chief:
+            summary_writer.add_summary(summary, step)
+
+      if sv.should_stop() or step >= args.steps:
+        tf_feed.terminate()
+
+      if sv.is_chief:
+        print("{0} exporting saved_model to: {1}".format(datetime.now().isoformat(), args.export_dir))
+        signatures = {
+          tf.saved_model.signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY: {
+            'inputs': { 'input': x },
+            'outputs': { 'output': prediction },
+            'method_name': tf.saved_model.signature_constants.PREDICT_METHOD_NAME
+          },
+          'featurize': {
```

Comment:
Shall we define these signature names as constants and provide them to Transformers as default?